python/lsst/pipe/drivers/constructCalibs.py

```diff
@@ -435,18 +437,20 @@ def getOutputId(self, expRefList, calibId):
         outputId.update(calibId)
         return outputId
 
-    def getMjd(self, dataId, timescale=dafBase.DateTime.UTC):
+    def getMjd(self, butler, dataId, timescale=dafBase.DateTime.UTC):
         """Determine the Modified Julian Date (MJD; in TAI) from a data identifier"""
-        dateObs = dataId[self.config.dateObs]
-
+        if self.config.dateObs in dataId:
+            dateObs = dataId[self.config.dateObs]
+        else:
+            dateObs = butler.queryMetadata('raw', ['dateObs'], dataId)[0]
```

Comment:
Shouldn't this be `butler.queryMetadata('raw', [self.config.dateObs], dataId)[0]`?
i.e., use the field we're told to, rather than hard-coding `dateObs`.